app/src/main/java/com/asfoundation/wallet/ui/iab/AppcoinsRewardsBuyPresenter.kt

```diff
@@ -127,26 +126,25 @@ class AppcoinsRewardsBuyPresenter(private val view: AppcoinsRewardsBuyView,
     disposables.add(
         appcoinsRewardsBuyInteract.isWalletBlocked()
             .subscribeOn(networkScheduler)
-            .observeOn(viewScheduler)
             .observeOn(networkScheduler)
             .flatMap { blocked ->
               if (blocked) {
                 appcoinsRewardsBuyInteract.isWalletVerified()
                     .observeOn(viewScheduler)
                     .doOnSuccess {
-                      if (it) view.showGenericError()
+                      if (it) view.showError(mapError(RewardPayment.Status.FORBIDDEN))
```

Comment:
Do we need the mapper here?  Why not have the showError receive the string directly? Also the other genericErrors maybe we should change them to showError with unknown error string